src/components/agents/q_agent.jl

```diff
@@ -0,0 +1,39 @@
+export AbstractQAgent, DQN
+
+abstract type AbstractQAgent <: AbstractAgent end
+
+selector(agent::AbstractQAgent) = agent.selector
+
+function (agent::AbstractQAgent)(obs::EnvObservation)
+    obs |> state |> learner(agent) |> selector(agent)
+end
+
+mutable struct DQN{Tl<:QLearner{<:NeuralNetworkQ}, Tb<:AbstractTurnBuffer, Ts<:AbstractDiscreteActionSelector} <: AbstractQAgent
+    role::String
+    learner::Tl
+    buffer::Tb
+    selector::Ts
+    batch_size::Int
+    update_horizon::Int  # starts with 1
+    γ::Float64
+    act_step::Int
+    min_replay_history::Int
+end
```

Comment:
I always thought of `batch_size`, `update_horizon`, `γ` and `min_replay_history` as parameters of the learner, since these parameters affect learning. I think the current concept `Actor` is not yet entirely clear to me...

Reply:
Yeah, I also had the same thought before. I'll explain the decision in detail soon.

Reply:
I have been thinking of this comment for a lot of times this week. Now I would like to retract my comment. Fortunately, it's not too late... 😃 

It's true that it would be more intuitive to move those parameters in the learner. This change will be reflected in the latest PR.

Thanks again for your great thoughts!

Reply:
Nice!